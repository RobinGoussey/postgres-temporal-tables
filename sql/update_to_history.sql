CREATE EXTENSION IF NOT EXISTS psql_temporal;

CREATE TABLE update_to_history_firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO update_to_history_firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('update_to_history_firemen_temporal');

-- test functionality of history
UPDATE update_to_history_firemen_temporal 
SET "rank" = 'Cpt.'
WHERE "rank" = 'Sgt.';

-- We can't use the timestamp columns, as they are constantly changing
select id , name , rank from temporal_update_to_history_firemen_temporal_history;

select id , name , rank from temporal_update_to_history_firemen_temporal_history where sysendtime is null;

-- Should be 2, since the start time should have been updated
select count(DISTINCT sysStartTime) from temporal_update_to_history_firemen_temporal_history;