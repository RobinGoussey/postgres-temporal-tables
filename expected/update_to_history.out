CREATE EXTENSION IF NOT EXISTS psql_temporal;
NOTICE:  extension "psql_temporal" already exists, skipping
CREATE TABLE update_to_history_firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO update_to_history_firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('update_to_history_firemen_temporal');
NOTICE:  trigger "update_to_history_firemen_temporal_temporal_trigger" for relation "public.update_to_history_firemen_temporal" does not exist, skipping
 convert_to_temporal 
---------------------
 t
(1 row)

-- test functionality of history
UPDATE update_to_history_firemen_temporal 
SET "rank" = 'Cpt.'
WHERE "rank" = 'Sgt.';
-- We can't use the timestamp columns, as they are constantly changing
select id , name , rank from temporal_update_to_history_firemen_temporal_history;
 id | name | rank 
----+------+------
  1 | Bram | Sgt.
  1 | Bram | Cpt.
(2 rows)

select id , name , rank from temporal_update_to_history_firemen_temporal_history where sysendtime is null;
 id | name | rank 
----+------+------
  1 | Bram | Cpt.
(1 row)

